### DISCUSSION TITLE
How to remove the white space on the left side and right side when i want to use zxing-python to genetate Code128 pictures?

### DISCUSSION THREAD

**Discussion by Vdeilae** [zxing-cpp/zxing-cpp#627] [Q&A]
i tried to set "quiet_zone" to 0,but it changed nothing ,the white space still exists.here is my code.

`img0 = zxingcpp.write_barcode(zxingcpp.BarcodeFormat.Code128, PART, width=200, height=30,quiet_zone=0,ec_level=-1)`

**Comment by axxel** [comment]
The "problem" is that you specified a width of 200 pixels. So the writer will generate an image buffer that is 200 pixels wide. If the symbol itself is originally e.g. 80 pixels, then it will be scaled to the 160 and the remaining 40 pixels will be white.

To generate a symbol with no white space, pass `width=0, quiet_zone=0`. Then you will get an image of 80 pixels width in the above example. If you then want to have it larger, scale it yourself. Make sure to use an integer scaling factor or to scale it at least 5x, otherwise you will likely destroy the readability (i.e. 5.5x should still work, while 1.5x will definitely not).
